### PR TITLE
support tuning step parameter range parameterization + support retry strategy in tuner

### PR DESCRIPTION
**Pull request by jerrypeng7773**
*Issue #, if available:*

*Description of changes:*

1. support tuning step tuning configuration parameter range parameterization
2. support retry strategy for estimator used by tuner 

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/tuner.py [hunk 1]
```diff
@@ -1580,6 +1583,9 @@ def _prepare_training_config(
         if parameter_ranges is not None:
             training_config["parameter_ranges"] = parameter_ranges
 
+        if estimator.max_retry_attempts is not None:
+            training_config["max_retry_attempts"] = estimator.max_retry_attempts
```
Comment (danabens) [inline src/sagemaker/tuner.py:1587]: do we already have a standard mechanism to propagate API changes to pipeline steps? i.e. simultaneous release of new API features and pipeline support

**Review by danabens — APPROVED** [review]

File: src/sagemaker/tuner.py (hunk 1 above)
Comment (jerrypeng7773) [reply src/sagemaker/tuner.py:1587]: from DSL level yes, as long as the model has it, we can support it simultaneously. However, in python SDK, when translating the estimator or tuner to request arguments, one will need to manually add the new fields. For instance, in this case, the estimator supports `max_retry_attempts`, however when passed to tuner, and during python object to request arguments translations, this is not included.

File: tests/integ/test_workflow.py [hunk 2]
```diff
@@ -1098,14 +1098,17 @@ def test_tuning(
         role=role,
         framework_version="1.5.0",
         py_version="py3",
-        instance_count=1,
-        instance_type="ml.m5.xlarge",
+        instance_count=instance_count,
```
Comment (ahsan-z-khan) [inline tests/integ/test_workflow.py:1101]: Where are these coming from? Do we have fixtures for `instance_count` and `instance_type` ?

Comment (jerrypeng7773) [reply tests/integ/test_workflow.py:1101]: line 1093 and 1094, they are the pipeline parameters

**Review by ahsan-z-khan — APPROVED** [review]
LGTM!

**Pull request merged**